client/Hooks/useLogger/Hook.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import { useEffect, useContext } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import { LogLevelType, LoggingContext } from 'Contexts';
+
+export const MESSAGE_BUFFER_MAX_SIZE = 100;
+
+export type LoggerType = {
+  log: (clientLevel: LogLevelType, msg: string) => void;
+  fatal: (msg: string) => void;
+  error: (msg: string) => void;
+  warn: (msg: string) => void;
+  info: (msg: string) => void;
+  debug: (msg: string) => void;
+  trace: (msg: string) => void;
+};
+
+// Unique client ID created when this module is loaded
+const clientID = uuidv4();
+
+/**
+ * Checks if the component should be logged by checking if the LOGGING
+ * query parameter regex metches the component name.
+ *
+ * @param componentName the component name to check
+ */
+const shouldLogComponent: (componentName: string) => boolean = (
+  componentName: string
+) => {
```

Comment:
You can streamline typing like this:

```suggestion
const shouldLogComponent = (
  componentName: string
): boolean => {
```

then `shouldLogComponent` will infer the type from the right side. It saves having to write parameters and their types twice, which becomes painful when you change them

Reply:
Nice! Thanks for that, makes it much tidier :)